sys/src/9/amd64/multiboot.c

```diff
@@ -103,6 +105,21 @@ mbtypename(int type)
 	return "(unknown)";
 }
 
+static i32
+mbmoduleread(Chan *c, void *buf, i32 len, i64 off)
+{
+	if (!initrd) {
+		error("mbmoduleread: module not found");
+	}
+
+	u64 modlen = initrd->modend - initrd->modstart;
+	if (off < 0 || off > modlen) {
```

Comment:
I think readmem should do the test of off < 0, and I think it already does the test for off > size.
So it makes sense (to me) to make sure readmem does the tests on lines 116-118 and then you can skip them here? 
for now I guess this will do.